airflow/docker.go

```diff
@@ -595,7 +595,7 @@ func Deploy(path, name, wsId string, prompt bool) error {
 	}
 
 	image, tag := docker.GetImageTagFromParsedFile(cmds)
-	if config.CFG.ShowWarnings.GetBool() && image != messages.VALID_DOCKERFILE_BASE_IMAGE {
+	if config.CFG.ShowWarnings.GetBool() && !validImageRepo(image) {
 		i, _ := input.InputConfirm(fmt.Sprintf(messages.WARNING_INVALID_IMAGE_NAME, image, messages.VALID_DOCKERFILE_BASE_IMAGE))
```

Comment:
@andriisoldatenko are we going to keep the warning message the same, as in, referring to quay.io/...? OR should we say 'quay.io/... or astrononmerinc/ap-airflow'?

Reply:
Yeah good catch. I think it's make sense only show new registry. 
@paolaperaza do you think we should highlight both in error message?

Reply:
@vishwas-astro `
```
Authenticated to dev.astrodev.io

Select which airflow deployment you want to deploy to:
 #     LABEL                   DEPLOYMENT NAME              WORKSPACE     DEPLOYMENT ID
 1     new-deployment-name     newtonian-telescope-1353     w1            ckhggklqd1938261p8nmbl3r41y

> 1
Deploying: newtonian-telescope-1353
newtonian-telescope-1353/airflow
Building image...
WARNING! You are using an invalid image name 'ap-airflow' in your Dockerfile, please use 'astronomerinc/ap-airflow' or 'quay.io/astronomer/ap-airflow'. Are you sure you want to continue?
```
changed warn message ^

Reply:
I think we should say something like
`We notice that you are not pulling from our recommended repo $repo_name and are using $invalid_repo_name instead. We strongly recommend you to use our certified images in our platform. Are you sure you want to continue with the custom image, (y/n)?`

Reply:
cc @paolaperaza @schnie

Reply:
@vishwas-astro @andriisoldatenko catching up here. What triggers this message? What's the difference between the error a user gets if they push an unsupported image (say, some random 1.10.3 image) versus an image/tag we support but associated with the wrong registry/repo? I assume the correction we're making here is for users who call the wrong registry? And this would naturally happen if they have a custom airflow image?

Reply:
@paolaperaza when supported registry we show nothing, and only when custom registry/tag we show warning. I added second repo because right now people still trying to push valid registry `astronomerinc/ap-airflow` and we show warning because we compare it with new registry `quay.io/astronomer/ap-airflow`.

Reply:
Maybe -


> WARNING! The image in your Dockerfile is pulling from `$invalid_repo_name`, which is not supported. We strongly recommend that you use Astronomer Certified images that pull from `astronomerinc/ap-airflow` or `quay.io/astronomer/ap-airflow`. If you're running a custom image, you can override this. Are you sure you want to continue?